### ISSUE TITLE
The field current_password of Set Username endpoint is not required as it should be

### ISSUE THREAD

**Issue by truonghn** [sunscrapers/djoser#479]
Hi Djoser team, Thank you very much for creating such an amazing library. I am having an issue when using Djoser. That is when I used this endpoint: https://djoser.readthedocs.io/en/2.0.3/base_endpoints.html#set-username The `current_password` field is not required as it should be. We could update email without providing the current_password.
if this makes sense, could I submit a PR to add required to the appropriate field?

**Comment by adriangzz** [comment]
I'm having the same issue on Django 3.0.2, djoser 2.0.3 and django 3.8.

**Comment by krish1237** [comment]
I'm having the same issue

**Comment by krish1237** [comment]
Hi, I debugged the source code pertaining the /set_username . It seems to be working fine. Try to reinstall djoser and see if it works. It worked for me that way.

**Comment by truonghn** [comment]
Thank you for your information @krish1237 .
If @krish1237 solution does not work with you, you could try this and see if it works @adriangzz :
So in sub class that inherited from `SetUsernameSerializer`, I added this:
```
class AccountSetUsernameSerializer(SetUsernameSerializer):

    def __init__(self, *args, **kwargs):
        self.fields["current_password"] = serializers.CharField(
            style={"input_type": "password"}
        )
        super().__init__(*args, **kwargs)
```
The reason is in Meta attribute of  `UsernameSerializer` class, it overridden field `fields`, so the field `current_password` was not validated.

**Comment by krish1237** [comment]
```
class CurrentPasswordSerializer(serializers.Serializer):
    default_error_messages = {
        "invalid_password": settings.CONSTANTS.messages.INVALID_PASSWORD_ERROR
    }

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields["current_password"] = serializers.CharField(style={"input_type": "password"})

    def validate_current_password(self, value):
        is_password_valid = self.context["request"].user.check_password(value)
        if is_password_valid:
            return value
        else:
            self.fail("invalid_password")
```
This should be the code of CurrentPasswordSerializer for all components extending this class to work as they should. This is because django only considers first extended Serializer as a base class i.e self and any new fields added with ```serializers.Field``` are discarded

**Comment by truonghn** [comment]
That makes perfect sense, @krish1237 . The reason I did not add this to `CurrentPasswordSerializer` class is I don't want to update the code of core lib :) before we confirmed this is the best approach :) . Thanks for your help.

**Comment by krish1237** [comment]
Updating the code of core lib is never a good approach. Your approach is a better fix for now but you have to add a  separate url endpoint that means two endpoints exist now. /users/me/ and /path/to/AccountSetUsername/

I have raised a pull request with the updated code. Lets see what happens. It should work with all existing tests

**Comment by truonghn** [comment]
@krish1237 , yeah, I had this config on the setting file so I don't need to have two endpoints, just FYI:
```
DJOSER = {
    'SERIALIZERS': {
        'set_username': 'accounts.serializers.AccountSetUsernameSerializer',
    }
}
```
